packages/plugin-electron-session/session.js

```diff
@@ -11,7 +11,10 @@ module.exports = (app, BrowserWindow) => ({
       return
     }
 
-    app.whenReady().then(() => { client.startSession() })
+    // jump through hoops to ensure the session gets started _after_ we have a device id
+    filestore.getDeviceInfo().then(() => {
+      app.whenReady().then(() => { setTimeout(() => client.startSession(), 10) })
+    }).catch(() => {})
```

Comment:
This might be a bit cleaner looking with `Promise.all`; it might change the timing though

```suggestion
    Promise.all([filestore.getDeviceInfo(), app.whenReady()])
      .then(() => { setTimeout(() => client.startSession(), 10) })
      .catch(() => {})
```

Reply:
Nice, let's give it a go!